sktime/forecasting/arch/_uarch.py

```diff
@@ -0,0 +1,362 @@
+        update_freq iterations. Set to 0 to disable iterative output.
+    disp : {bool, "off", "final"}
+        Either 'final' to print optimization result or 'off' to display
+        nothing. If using a boolean, False is "off" and True is "final"
+    starting_values : np.ndarray, optional
+        Array of starting values to use.  If not provided, starting values
+        are constructed by the model components.
+    cov_type : str, optional
+        Estimation method of parameter covariance.  Supported options are
+        'robust', which does not assume the Information Matrix Equality
+        holds and 'classic' which does.  In the ARCH literature, 'robust'
+        corresponds to Bollerslev-Wooldridge covariance estimator.
+    show_warning : bool, optional
+        Flag indicating whether convergence warnings should be shown
+    first_obs : {int, str, datetime, Timestamp}
+        First observation to use when estimating model
+    last_obs : {int, str, datetime, Timestamp}
+        Last observation to use when estimating model
+    tol : float, optional
+        Tolerance for termination.
+    options : dict, optional
+        Options to pass to scipy.optimize.minimize.  Valid entries
+        include 'ftol', 'eps', 'disp', and 'maxiter'.
+    backcast : {float, np.ndarray}, optional
+        Value to use as backcast. Should be measure \\sigma^2_0
+        since model-specific non-linear transformations are applied to
+        value before computing the variance recursions.
+    params : {np.ndarray, Series}
+        Parameters required to forecast. Must be identical in shape to the
+        parameters computed by fitting the model.
+    start : {int, datetime, Timestamp, str}, optional
+        An integer, datetime or str indicating the first observation to produce the
+        forecast for. Datetimes can only be used with pandas inputs that have a
+        datetime index. Strings must be convertible to a date time, such as in
+        '1945-01-01'.
+    align : str, optional
+        Either 'origin' or 'target'. When set of 'origin', the t-th row of forecast
+        contains the forecasts for t+1, t+2, ..., t+h. When set to 'target', the
+        t-th row contains the 1-step ahead forecast from time t-1, the 2 step from
+        time t-2, ..., and the h-step from time t-h. 'target' simplified computing
+        forecast errors since the realization and h-step forecast are aligned.
+    method : {'analytic', 'simulation', 'bootstrap'}
+        Method to use when producing the forecast. The default is analytic.
+        The method only affects the variance forecast generation. Not all volatility
+        models support all methods.
+        In particular, volatility models that do not evolve in squares such as
+        EGARCH or TARCH
+        do not support the 'analytic' method for horizons > 1.
+    simulations : int
+        Number of simulations to run when computing the forecast using either
```

Comment:
These arguments seem to suggest support for probabilistic forecasts, and package documentation confirms it. Do you want to add support for that?

Reply:
I shall look into it.

Reply:
I need some more time, maybe I can add them in my next PR.